src/model.ts

```diff
@@ -462,3 +463,61 @@ export class SpecQueryModel {
     this._rankingScore[rankingName] = score;
   }
 }
+
+export class SpecQueryModelGroup {
+  private _name: string;
+  private _path: string;
+  private _items: (SpecQueryModel | SpecQueryModelGroup)[];
+  private _groupBy: GroupBy;
+  private _orderGroupBy: string;
+
+  constructor(name: string = '', path: string = '', items: (SpecQueryModel | SpecQueryModelGroup)[] = [],
+              groupBy: GroupBy = undefined, orderGroupBy: string = undefined) {
+    this._name = name;
+    this._path = path;
+    this._items = items;
+    this.groupBy = groupBy;
+    this._orderGroupBy = orderGroupBy;
+  }
+
+  public getTopSpecModel(g: SpecQueryModelGroup): SpecQueryModel {
+    const topItem = g._items[0];
+    if (topItem instanceof SpecQueryModelGroup) {
+      return this.getTopSpecModel(topItem);
+    } else {
+      return topItem;
+    }
+  }
+
+  public get name() {
+    return this._name;
+  }
+
+  public set name(name: string) {
+    this._name = name;
+  }
+
+  public get items() {
+    return this._items;
+  }
+
+  public set items(items: (SpecQueryModel | SpecQueryModelGroup)[]) {
+    this._items = items;
+  }
+
+  public set groupBy(groupBy: GroupBy) {
+    this._groupBy = groupBy;
+  }
+
+  // Currently, we do not need to get orderGroupBy for anything in CQL.
```

Comment:
Have you check if we use them in editor / vlui? 

It's kinda weird if you can set something but can't get them.  Usually it's the other way around.  (Can `get` but not `set`)